### PR TITLE
fix loading single column of data

### PR DESCRIPTION
**Pull request by gregllong**
Added a few lines to take care of single column of data.

Also added a test and a fixture file for testing loading of single column of data.

### CODE REVIEW

File: _data.py [hunk 1]
```diff
@@ -287,7 +287,11 @@ def fix(x): return x.replace('i','j')
 
         # genfromtxt returns a 1D array if there is only one data line.
         # highly confusing behavior, numpy!
-        if len(_n.shape(z)) == 1: z = _n.array([z])
+        if len(_n.shape(z)) == 1:
+            # check to make sure the data file contains only 1 column of data
+            total_data_line_number = len(lines) - first_data_line
```
Comment (gregllong) [inline _data.py]: that's probably a better name for that variable.

**Comment by jayich** [comment]
@gregllong , I would make the change to "rows_of_data".  LGTM.

**Comment by jaxankey** [comment]
Make sure to add a test and that it doesn't break the auto-delimiter detection.

**Comment by gregllong** [comment]
I will do that as soon as I finish teaching today.

**Comment by jayich** [comment]
@gregllong , if you add this CSV file test I would be much obliged.

**Comment by gregllong** [comment]
@jayich could you add the file that you failed loading? I can come up with a fake one, but it's probably easier if I could use the one you used.

**Comment by jayich** [comment]
@gregllong , fixture is in this commit, try plotting this file and you will see the error.

https://github.com/Spinmob/spinmob/commit/42577ec3f806d7ba05ca74a267d5659451107f25

**Comment by gregllong** [comment]
@jayich what is the software that created this .dat file? When I do readline() with python I see a whole bunch of "\x00" in the last line although when you open it with a text editor you just see a blank line. That's probably why the file fails to load.

**Comment by jayich** [comment]
@gregllong , it was made with a Raspberry Pi.  That is a null hex character, see here: http://stackoverflow.com/questions/1182812/what-is-the-meaning-of-x00-x04-in-php

Perhaps this could be treated like white space in spinmob?  Probably ramifications outside of this use case seem small?  @jaxankey , any thoughts?

I'm happy either way, leave the code as is, or decide that this is something that we should "fix".

**Comment by gregllong** [comment]
I branched off an earlier master version and seems like it would not load the file using ```spinmob.data.load()``` either. To open the file I used the following code, copied from load_file()  in databox:

```python
>>> path = sm.dialogs.open_single()
>>> f = open(path, 'uU')
>>> lines = f.readlines()
>>> f.close()
>>> print lines
```

I can add a few more steps to the load_file() method in Databox, but I'm not sure if @jaxankey would like me to do that because that just seems to clump the code even more. 

I'm going to switch gears and see what I can do with autodetecting delimiter test.

**Comment by jaxankey** [comment]
Didn't understand that last message.

Is there an option in python's open / readlines that either ignores the
\x00 (null) character or something? Does string.strip() remove nulls?

Could go this route on each line, prior to splitting:

>>> import string
>>> s.strip(string.whitespace+'\x00')

**Comment by gregllong** [comment]
Sorry for not being clear. I felt I shouldn't add strip(string.whitespace+'\x00') to every strip() because right now in the fixture file '\x00' only shows up in the last line. There are a few places with strip() and I am not confident to change all of them without testing them well.  

I vote for leaving the code as it is , and maybe just delete the last empty line of the file.

**Comment by jaxankey** [comment]
I don't. :)

Can you define a constant at the top that is string.whitespace+'\x00', then
use that string as the argument for all the strips? We have tests in place,
so it shouldn't be hard to troubleshoot.

Recommend using an _ variable like "_whitespace" so the spinmob.data.
namespace does not get cluttered (one goal is to have it be easily
navigated by a user on the command line).

-Jack

**Comment by jayich** [comment]
Nice, I like pushing the improvements.  I was had thoughts along the same lines regarding our tests.

**Comment by jaxankey** [comment]
"I was had"? Aren't you a professor?

-Jack

Sent from sender's phone with typos. Sender not liable for inflammatory
autocorrects.

**Pull request merged**